validator/src/journal/commit_store.rs

```diff
@@ -329,13 +329,12 @@ impl CommitStore {
                 block
                     .batches
                     .into_iter()
-                    .skip_while(|batch| {
-                        batch
+                    .find(|batch| {
```

Comment:
@scealiontach, does this logic look right to you? I fixed the clippy warning here using the original logic, but I was confused, why it's skipping each batch where any of its contained tx_ids match `transaction_id`.